### ISSUE TITLE
Backporting

### ISSUE THREAD

**Issue by Tyrannicodin** [Trivaxy/Tiny-Item-Animations#1]
I've been taking a look at backporting the mod as I'd like to use it for 1.18.2, unfortunately it seems difficult so far as there is no onItemSlotStack event before this patch (I found this line in the 1.19.3 recommended changelog):
 - 43.1.41 [1.19.x] Add event for items being stacked or swapped in a GUI. (#9050)
 
 I think I can work on a workaround, if you see one before I make a pull request please let me know.

**Comment by Trivaxy** [comment]
I assumed it was also there in the 1.18 version, but it's no problem - the fabric version uses a mixin to replace that event. I'm planning on doing the same thing for the backport which I'll get to either today or tomorrow.

**Comment by Tyrannicodin** [comment]
Yeah, that's the solution I went with - I'll make a push to my fork of this repository and you can see my solution and check it.

**Comment by Tyrannicodin** [comment]
Unfortunately I kept getting an error about the config files for the mixins, I’ve not worked with them lots so any insight you could provide would be appreciated.

**Comment by Trivaxy** [comment]
What's the error about?

**Comment by Tyrannicodin** [comment]
Exception in thread "main" org.spongepowered.asm.launch.MixinInitialisationError: Error initialising mixin config tia.mixins.json
And
Caused by: java.lang.IllegalArgumentException: The specified resource 'tia.mixins.json' was invalid or could not be read
Seem to be the most important lines

**Comment by Trivaxy** [comment]
That's odd. Make sure tia.mixins.json is actually in the proper place and that the json isn't messed up. Try generating your run configurations again, just in case.

I need to head off for now - once I'm up, I'll take a crack at backporting it myself and letting you know how it goes - it should just come down to copying the fabric mixin over to forge 1.18, and backporting to fabric 1.18 which should be even simpler

**Comment by Tyrannicodin** [comment]
Re-ran my run configurations and checked the json, no change in the error message. I'll look further into it tomorrow as it's getting late for me now.

**Comment by Trivaxy** [comment]
I've successfully backported the mod to forge 1.18. I ran into your same issue, but the cause was simple - we forgot to alter the mods.toml file to bump down the required forge version 😅 . The error is really mysterious otherwise, since it doesn't mention anything of the sort.

I'll backport the fabric version as well and publish everything. Thanks a lot for taking the initiative to backport it though - I really appreciate it!
